File: comu/com.energyict.mdc.firmware/src/main/java/com/energyict/mdc/firmware/FirmwareStatus.java
/*
 * Copyright (c) 2017 by Honeywell International Inc. All Rights Reserved
 */

package com.energyict.mdc.firmware;

public enum FirmwareStatus {
    // The order defines the order of statuses & related firmware check options in UI
    TEST("test"),
    FINAL("final"),
    GHOST("ghost"),
    DEPRECATED("deprecated");

    private final String status;

    FirmwareStatus(String status) {
        this.status = status;
    }

    public String getStatus() {
        return status;
    }
}

File: comu/com.energyict.mdc.firmware.extjs/src/main/web/js/firmware/src/model/FirmwareStatus.js
/*
 * Copyright (c) 2017 by Honeywell International Inc. All Rights Reserved
 */

Ext.define('Fwc.model.FirmwareStatus', {
    extend: 'Ext.data.Model',
    fields: [
        {name: 'id', type: 'string', useNull: true},
        {name: 'localizedValue', type: 'string', useNull: true}
    ]
});